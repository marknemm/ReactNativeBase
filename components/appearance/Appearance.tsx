import ScreenView from '@components/screen-view/ScreenView';
import { useGeneralStyles } from '@hooks/styles-hooks';
import { CheckBox, Text } from '@rneui/themed';
import { View } from 'react-native';
import type { AppearanceProps } from './Appearance.interfaces';

/**
 * Component that allows the user to select the appearance of the application.
 *
 * @param props The {@link AppearanceProps}.
 * @returns The {@link Appearance} component.
 */
const Appearance: React.FC<AppearanceProps> = ({
  appearance,
  disabled,
  onAppearanceChange,
}) => {
  const generalStyles = useGeneralStyles();

  return (
    <ScreenView>
      <Text style={generalStyles.form.label}>
        Appearance
      </Text>
      <View style={generalStyles.view.row}>
        <CheckBox
          title="Auto"
          checked={appearance === 'auto'}
          checkedIcon="circle-half-full"
          uncheckedIcon="circle-half-full"
          iconType="material-community"
          textStyle={appearance === 'auto' ? generalStyles.text.doubleUnderline : undefined}
          onPress={() => onAppearanceChange?.('auto')}
          disabled={disabled}
        />
        <CheckBox
          title="Light"
          checked={appearance === 'light'}
          checkedIcon="sun-o"
          uncheckedIcon="sun-o"
          textStyle={appearance === 'light' ? generalStyles.text.doubleUnderline : undefined}
          onPress={() => onAppearanceChange?.('light')}
          disabled={disabled}
        />
        <CheckBox
          title="Dark"
          checked={appearance === 'dark'}
          checkedIcon="moon-o"
          uncheckedIcon="moon-o"
          textStyle={appearance === 'dark' ? generalStyles.text.doubleUnderline : undefined}
          onPress={() => onAppearanceChange?.('dark')}
          disabled={disabled}
        />
      </View>
    </ScreenView>
  );
};

export default Appearance;
export type * from './Appearance.interfaces';
