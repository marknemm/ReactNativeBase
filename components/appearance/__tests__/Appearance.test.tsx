import Appearance from '@components/appearance/Appearance';
import AppProvider from '@test/contexts/app/AppProvider';
import { fireEvent, render, screen } from '@testing-library/react-native';

describe('<Appearance />', () => {
  it('calls onAppearanceChange when the appearance changes', () => {
    const onAppearanceChange = jest.fn();
    render(
      <Appearance appearance="auto" onAppearanceChange={onAppearanceChange} />,
      { wrapper: AppProvider }
    );

    fireEvent.press(screen.getByText('Light'));
    expect(onAppearanceChange).toHaveBeenCalledWith('light');

    fireEvent.press(screen.getByText('Dark'));
    expect(onAppearanceChange).toHaveBeenCalledWith('dark');

    fireEvent.press(screen.getByText('Auto'));
    expect(onAppearanceChange).toHaveBeenCalledWith('auto');
  });

  it('does not call onAppearanceChange when disabled', () => {
    const onAppearanceChange = jest.fn();
    render(
      <Appearance appearance="auto" onAppearanceChange={onAppearanceChange} disabled />,
      { wrapper: AppProvider }
    );

    fireEvent.press(screen.getByText('Light'));
    expect(onAppearanceChange).not.toHaveBeenCalled();

    fireEvent.press(screen.getByText('Dark'));
    expect(onAppearanceChange).not.toHaveBeenCalled();

    fireEvent.press(screen.getByText('Auto'));
    expect(onAppearanceChange).not.toHaveBeenCalled();
  });

  describe('snapshots', () => {
    it('renders correctly auto appearance', () => {
      render(
        <Appearance appearance="auto" />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly dark appearance', () => {
      render(
        <Appearance appearance="dark" />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly light appearance', () => {
      render(
        <Appearance appearance="light" />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly disabled', () => {
      render(
        <Appearance appearance="auto" disabled />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
