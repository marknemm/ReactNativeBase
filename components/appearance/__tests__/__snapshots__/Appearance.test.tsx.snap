// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Appearance /> snapshots renders correctly auto appearance 1`] = `
<RCTScrollView
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
  testID="rnb-screen-scroll-container-view"
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "marginHorizontal": "auto",
          "maxWidth": 1200,
          "paddingBottom": 37.714285714285715,
          "paddingLeft": 18.857142857142858,
          "paddingRight": 18.857142857142858,
          "paddingTop": 37.714285714285715,
          "width": "100%",
        }
      }
      testID="rnb-screen-view"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Appearance
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": true,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="circle-half-full"
          collapsable={false}
          focusable={true}
          iconType="material-community"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="circle-half-full"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#007aff",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              󱎖
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "double",
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Auto
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="sun-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="sun-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Light
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="moon-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="moon-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Dark
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<Appearance /> snapshots renders correctly dark appearance 1`] = `
<RCTScrollView
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
  testID="rnb-screen-scroll-container-view"
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "marginHorizontal": "auto",
          "maxWidth": 1200,
          "paddingBottom": 37.714285714285715,
          "paddingLeft": 18.857142857142858,
          "paddingRight": 18.857142857142858,
          "paddingTop": 37.714285714285715,
          "width": "100%",
        }
      }
      testID="rnb-screen-view"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Appearance
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="circle-half-full"
          collapsable={false}
          focusable={true}
          iconType="material-community"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="circle-half-full"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              󱎖
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Auto
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="sun-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="sun-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Light
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": true,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="moon-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="moon-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#007aff",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "double",
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Dark
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<Appearance /> snapshots renders correctly disabled 1`] = `
<RCTScrollView
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
  testID="rnb-screen-scroll-container-view"
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "marginHorizontal": "auto",
          "maxWidth": 1200,
          "paddingBottom": 37.714285714285715,
          "paddingLeft": 18.857142857142858,
          "paddingRight": 18.857142857142858,
          "paddingTop": 37.714285714285715,
          "width": "100%",
        }
      }
      testID="rnb-screen-view"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Appearance
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": true,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="circle-half-full"
          collapsable={false}
          focusable={true}
          iconType="material-community"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="circle-half-full"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#007aff",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              󱎖
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "hsl(208, 8%, 63%)",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "double",
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Auto
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="sun-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="sun-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "hsl(208, 8%, 63%)",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Light
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="moon-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="moon-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "hsl(208, 8%, 63%)",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Dark
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<Appearance /> snapshots renders correctly light appearance 1`] = `
<RCTScrollView
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
  testID="rnb-screen-scroll-container-view"
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "marginHorizontal": "auto",
          "maxWidth": 1200,
          "paddingBottom": 37.714285714285715,
          "paddingLeft": 18.857142857142858,
          "paddingRight": 18.857142857142858,
          "paddingTop": 37.714285714285715,
          "width": "100%",
        }
      }
      testID="rnb-screen-view"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#86939e",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Appearance
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="circle-half-full"
          collapsable={false}
          focusable={true}
          iconType="material-community"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="circle-half-full"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              󱎖
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Auto
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": true,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="sun-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="sun-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#007aff",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "double",
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Light
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          checkedIcon="moon-o"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          replaceTheme={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 10,
            }
          }
          testID="RNE__CheckBox__Wrapper"
          uncheckedIcon="moon-o"
          updateTheme={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#bfbfbf",
                    "fontSize": 24,
                  },
                  {
                    "minWidth": 24,
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__Checkbox__Icon"
            >
              
            </Text>
            <Text
              accessibilityRole="text"
              style={
                {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="RNE__CheckBox__Title"
            >
              Dark
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
